builtin/logical/consul/path_token.go

```diff
@@ -103,13 +106,41 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 		})
 	}
 
+	aclServiceIdentities := []*api.ACLServiceIdentity{}
+	parsedServiceIdentities, err := parseSemicolonStringSlice(roleConfigData.ServiceIdentities)
+	if err != nil {
+		return nil, err
+	}
+	for _, serviceIdentity := range parsedServiceIdentities {
+		entry := &api.ACLServiceIdentity{}
+		components := strings.SplitN(serviceIdentity, ":", 2)
+		entry.ServiceName = components[0]
+		if len(components) == 2 {
+			entry.Datacenters = strings.Split(components[1], ",")
+		}
```

Comment:
nit: I think this could be simplified to only call `Split` once. Maybe:

```suggestion
		components := strings.Split(serviceIdentity, ":")
		entry.ServiceName = components[0]
		if len(components) > 1 {
			entry.Datacenters = components[1:]
		}
```

Reply:
I'm trying to think if this creates the same result.
If the provided value is "serv1:dc1,dc2,dc3" then the result is
```
ServiceName: "serv1"
Datacenters: [dc1 dc2 dc3]
```

I think this would result in Datacenters still being a string containing commas, or am I mistaken?

Reply:
Oh, I'm mistaken. I thought it was splitting twice on ":", but now that I read it, there are two separate split characters. My bad!